crates/sui-bcs-json/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+};
+
+fn to_json<T: serde::Serialize + for<'a> serde::Deserialize<'a>>(
+    data: Vec<u8>,
+) -> Result<serde_json::Value, Error> {
+    let bcs: T = bcs_from_bytes(&data)?;
+    Ok(json!(bcs))
+}
+
+fn to_bcs<T: serde::Serialize + for<'a> serde::Deserialize<'a>>(
+    data: &str,
+) -> Result<Vec<u8>, Error> {
+    let bcs: T = serde_json::from_str(data)?;
+    Ok(bcs::to_bytes(&bcs)?)
+}
+
+fn internal_bcs_to_json(typename: &str, data: Vec<u8>) -> Result<String, Error> {
+    let json = match typename {
+        "u8" => to_json::<u8>(data),
+        "u16" => to_json::<u16>(data),
+        "u32" => to_json::<u32>(data),
+        "u64" => to_json::<u64>(data),
+        "u128" => to_json::<u128>(data),
+        "bool" => to_json::<bool>(data),
+        "Argument" => to_json::<Argument>(data),
+        "CallArg" => to_json::<CallArg>(data),
+        "Command" => to_json::<Command>(data),
+        "GasData" => to_json::<GasData>(data),
+        "ObjectDigest" => to_json::<ObjectDigest>(data),
+        "ObjectID" => to_json::<ObjectID>(data),
+        "ProgrammableTransaction" => to_json::<ProgrammableTransaction>(data),
+        "SequenceNumber" => to_json::<SequenceNumber>(data),
+        "SuiAddress" => to_json::<SuiAddress>(data),
+        "TransactionData" => to_json::<TransactionData>(data),
+        unsupported => bail!("Requested type {} is currently unsupported.", unsupported),
+    };
+
+    if let Ok(json) = json {
+        Ok(serde_json::to_string_pretty(&json)?)
+    } else {
+        let error_msg = json.err();
+        bail!("Cannot convert bcs to json, {}", error_msg.unwrap())
+    }
+}
+
+fn internal_json_to_bcs(typename: &str, data: &str) -> Result<Vec<u8>, Error> {
+    let bcs_bytes = match typename {
+        "u8" => to_bcs::<u8>(data),
+        "u16" => to_bcs::<u16>(data),
+        "u32" => to_bcs::<u32>(data),
+        "u64" => to_bcs::<u64>(data),
+        "u128" => to_bcs::<u128>(data),
+        "bool" => to_bcs::<bool>(data),
+        "Argument" => to_bcs::<Argument>(data),
+        "CallArg" => to_bcs::<CallArg>(data),
+        "Command" => to_bcs::<Command>(data),
+        "GasData" => to_bcs::<GasData>(data),
+        "ObjectDigest" => to_bcs::<ObjectDigest>(data),
+        "ObjectID" => to_bcs::<ObjectID>(data),
+        "ProgrammableTransaction" => to_bcs::<ProgrammableTransaction>(data),
+        "SequenceNumber" => to_bcs::<SequenceNumber>(data),
+        "SuiAddress" => to_bcs::<SuiAddress>(data),
+        "TransactionData" => to_bcs::<TransactionData>(data),
+        unsupported => bail!("Requested type {} is currently unsupported.", unsupported),
+    };
+
+    if let Ok(bcs_bytes) = bcs_bytes {
+        Ok(bcs_bytes)
+    } else {
+        bail!("Cannot convert json to bcs, {}", bcs_bytes.err().unwrap())
+    }
+}
+
+/// Returns a BCS array (u8) from the input json string.
+#[ffi_export]
+fn json_to_bcs(type_name: char_p::Ref<'_>, json_data: char_p::Ref<'_>) -> repr_c::Vec<u8> {
```

Comment:
C doesn't have namespacing so you will need to prefix these function names with something to make sure they don't conflict with other functions.  How about `sui_bcs_to_json` and `sui_bcs_from_json`?